File: Matlab/WarningType.m
function [ WarningName ] = WarningType( Number )
% This function returns the warning type by specifying the warning location
% IMPORTANT: Incase warnings have been added or removed from here, change
% WarningDefenitionScript and WarningWorthAlert
if Number == 1
    WarningName = 'Irregular Heart Rate';
elseif Number == 2
    WarningName = 'Abnornal P Wave';
elseif Number == 3
    WarningName = 'Abnornal R Wave';
elseif Number == 4
    WarningName = 'Abnornal T Wave';
elseif Number == 5
    WarningName = 'Long P Wave';
elseif Number == 6
    WarningName = 'High P Wave';
elseif Number == 7
    WarningName = 'High T Wave';
elseif Number == 8
    WarningName = 'Long QRS Segment';
elseif Number == 9
    WarningName = 'Short QRS Segment';
elseif Number == 10
    WarningName = 'Long PR Interval';
elseif Number == 11
    WarningName = 'Short PR Interval';
elseif Number == 12
    WarningName = 'Long QTc Interval';
elseif Number == 13
    WarningName = 'Short QTc Interval';
elseif Number == 14
    WarningName = 'ST Elevation';
elseif Number == 15
    WarningName = 'ST Depression'; 
elseif Number == 16
    WarningName = 'Significant Q wave';
elseif Number == 17
    WarningName = 'Inverted T wave';
elseif Number == 18
    WarningName = 'Fast Heart Rate during the morning';
elseif Number == 19
    WarningName = 'Myocardial Infaction';
elseif Number == 20
    WarningName = 'Significant S wave';
elseif Number == 21
    WarningName = 'Low P Wave';
elseif Number == 22
    WarningName = 'Low Q Wave';
elseif Number == 23
    WarningName = 'Missed Q Wave';
elseif Number == 24
    WarningName = 'Missed P Wave';
% If the number specified is out of bound return an error message
else
   msg = 'Invalid Warning Number'; 
   error(msg);
end

File: Python Project Filter-Detect-GUI/warning_type.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 19 07:45:34 2017

@author: Mostafa Hammoud

This script contains the names of the warnings
"""

def warning_type(number):
    
    if number == 0:
        warning_name = 'Irregular Heart Rate'
    elif number == 1:
        warning_name = 'Abnornal P Wave'
    elif number == 2:
        warning_name = 'Abnornal R Wave'
    elif number == 3:
        warning_name = 'Abnornal T Wave'
    elif number == 4:
        warning_name = 'Long P Wave'
    elif number == 5:
        warning_name = 'High P Wave'
    elif number == 6:
        warning_name = 'High T Wave'
    elif number == 7:
        warning_name = 'Long QRS Segment'
    elif number == 8:
        warning_name = 'Short QRS Segment'
    elif number == 9:
        warning_name = 'Long PR Interval'
    elif number == 10:
        warning_name = 'Short PR Interval'
    elif number == 11:
        warning_name = 'Long QTc Interval'
    elif number == 12:
        warning_name = 'Short QTc Interval'
    elif number == 13:
        warning_name = 'ST Elevation'
    elif number == 14:
        warning_name = 'ST Depression'
    elif number == 15:
        warning_name = 'Significant Q wave'
    elif number == 16:
        warning_name = 'Inverted T wave'
    elif number == 17:
        warning_name = 'Fast Heart Rate during the morning'
    elif number == 18:
        warning_name = 'Myocardial Infaction'
    elif number == 19:
        warning_name = 'Significant S wave'
    elif number == 20:
        warning_name = 'Low P Wave'
    elif number == 21:
        warning_name = 'Low Q Wave'
    elif number == 22:
        warning_name = 'Missed Q Wave'
    elif number == 23:
        warning_name = 'Missed P Wave'
    elif number == 24:
        warning_name = 'Low S'
    # If the number specified is out of bound return an error message
    else:
        warning_name = []
        print('Invalid Warning Number') 
      
    return  warning_name



high_p       = warning_type(5)
high_t       = warning_type(6)
low_q        = warning_type(21)
low_p        = warning_type(20)
low_s        = warning_type(24)
sig_q        = warning_type(15)
sig_s        = warning_type(19)
inverted_t   = warning_type(16)
long_p       = warning_type(4)
long_qtc     = warning_type(11)
shirt_qrs    = warning_type(8)
st_elev      = warning_type(13)
irregular_hr = warning_type(0)

num_of_warnings = 25
